lib/rake/extensiontask.rb

```diff
@@ -158,26 +157,22 @@ def define_compile_tasks(for_platform = nil, ruby_ver = RUBY_VERSION)
       directory lib_path
       directory stage_binary_dir_path
 
-      directory File.dirname(siteconf_path)
-      # Set paths for "make install" destinations
-      file siteconf_path => File.dirname(siteconf_path) do
-        File.open(siteconf_path, "w") do |siteconf|
-          siteconf.puts "require 'rbconfig'"
-          siteconf.puts "require 'mkmf'"
-          siteconf.puts "dest_path = mkintpath(#{File.expand_path(lib_path).dump})"
-          %w[sitearchdir sitelibdir].each do |dir|
-            siteconf.puts "RbConfig::MAKEFILE_CONFIG['#{dir}'] = dest_path"
-            siteconf.puts "RbConfig::CONFIG['#{dir}'] = dest_path"
-          end
-        end
-      end
-
       # copy binary from temporary location to final lib
       # tmp/extension_name/extension_name.{so,bundle} => lib/
       task "copy:#{@name}:#{platf}:#{ruby_ver}" => [lib_path, tmp_binary_path, "#{tmp_path}/Makefile"] do
         # install in lib for native platform only
         unless for_platform
-          sh "#{make} install target_prefix=", chdir: tmp_path
+          expaned_lib_path = mkintpath(File.expand_path(lib_path))
```

Comment:
Can we use relative path instead like RubyGems did https://github.com/rubygems/rubygems/pull/5626 ?

```suggestion
         relative_lib_path = Pathname(lib_path).relative_path_from(tmp_path)
```

I don't want to require `mkmf`.

Reply:
I have update the `sitearchdir` and `sitelibdir` to use the expanded path to be more debuggable :)

```ruby
expaned_lib_path = File.expand_path(lib_path)
```

ps. Thank you for reviewing the PR @kou !